app/src/main/java/net/gsantner/opoc/util/FileUtils.java

```diff
@@ -504,13 +505,9 @@ public static File join(File file, String... childSegments) {
         return file;
     }
 
-    public static String sha512sum(final byte[] bytes) {
+    public static String sha512sum(final byte[] data) {
         try {
-            final StringBuilder sb = new StringBuilder();
-            for (final byte b : MessageDigest.getInstance("SHA-512").digest(bytes)) {
-                sb.append(Integer.toString((b & 0xff) + 0x100, 16).substring(1));
-            }
-            return sb.toString();
+            return Arrays.toString(MessageDigest.getInstance("SHA-512").digest(data));
```

Comment:
Idiomatic and faster

Reply:
do we also use it? but yeah shorter and better

md5 is just fine as well - speed is way more interesting here than "maximum collision safety"

Reply:
switched back to crc32. It's the fastest and 1 in 2^32 collision chance is good enough.

Reply:
OK, hope that's OK for you, I think it's the better option.

btw the hash function takes a alg parameter, but currently uses hardcoded SHA-512

```suggestion
            return Arrays.toString(MessageDigest.getInstance(alg).digest(data));
```